src/java.base/share/classes/java/net/URLEncoder.java

```diff
@@ -138,6 +137,24 @@ public class URLEncoder {
         DEFAULT_ENCODING_NAME = StaticProperty.fileEncoding();
     }
 
+    private static void encodeByte(StringBuilder out, byte b) {
```

Comment:
I took a look at `HexFormat` and found that performance could be touched up a notch: #15591

Reply:
#15591 has been pushed, and when I test changing `encodeByte` to this:
```java
    private static void encodeByte(StringBuilder out, byte b) {
        out.append('%');
        HexFormat.of().withUpperCase().toHexDigits(out, b);
    }
```
... results are performance neutral:
```
Name                           (unchanged) Cnt  Base   Error   Test   Error  Unit   Diff%
URLEncodeDecode.testEncodeUTF8           0  15 2,069 ± 0,027  1,943 ± 0,004 ms/op    6,1% (p = 0,000*)
URLEncodeDecode.testEncodeUTF8          75  15 0,946 ± 0,025  0,920 ± 0,002 ms/op    2,7% (p = 0,001*)
URLEncodeDecode.testEncodeUTF8         100  15 0,484 ± 0,008  0,473 ± 0,016 ms/op    2,3% (p = 0,019 )
  * = significant
```
Feel free to sync up with master and use that, since it simplifies and consolidates the implementation. Either way you decide I can approve and sponsor this.